vanetza/security/naive_certificate_provider.hpp

```diff
@@ -48,6 +48,15 @@ class NaiveCertificateProvider : public CertificateProvider
      */
     const Certificate& root_certificate();
 
+    /**
+     * \brief generate a authorization ticket
+     *
+     * \param mutator allows mutating the certificate directly before signing, useful in tests
+     *
+     * \return generated certificate
+     */
+    Certificate generate_authorization_ticket(std::function<void(Certificate&)> mutator);
```

Comment:
Hm, this may be a little bit too complicated in the long run. Maybe it is sufficient (for testing) to provide a sign_certificate() method here which enables to forge malicious certificates with correct signature. In my opinion it would not hurt to calculate the signature twice for testing.

Reply:
Yes, calculating the signature twice totally doesn't hurt, I'll change that.

Reply:
Changed.